storage/s3/options.go

```diff
@@ -290,18 +402,8 @@ func (opts *Options) toS3Options() (
 		expires = *opts.DefaultExpire
 	}
 	presignOpts = func(s3Opts *s3.PresignOptions) {
+		opts.presignOptions(s3Opts)
 		s3.WithPresignExpires(expires)(s3Opts)
-		if opts.ExternalURI != nil {
-			presignURL := *opts.ExternalURI
-			resolver := s3.EndpointResolverFromURL(presignURL,
-				func(ep *aws.Endpoint) {
-					ep.HostnameImmutable = opts.ForcePathStyle
-				},
-			)
-			s3.WithPresignClientFromClientOptions(
-				s3.WithEndpointResolver(resolver),
```

Comment:
again showing that I am reading: presignOptions added `s.options` here

Reply:
Yes, they were moved from inside `SimpleStorageService.*Request` interface to the `Options.preSignOptions` function to lower code repeatability.